app/celery/provider_tasks.py

```diff
@@ -62,6 +63,21 @@ def check_sms_delivery_receipt(self, message_id, notification_id, sent_at):
                 provider_response=provider_response,
             )
             raise self.retry(exc=ntfe)
+        except ClientError as err:
+            # Probably a ThrottlingException but could be something else
```

Comment:
Should we check for the `ThrottlingException` explicitly because that's a known case, and retain the rest in the catchall `ClientError` for more anything else, or is the error handling the same regardless and therefore this is fine?

Reply:
I think regardless of what type of exception it is we want to keep retrying because:

1.  If we retry, at some point it may work.
2. Even if it doesn't work, getting 48 exceptions plus a bunch of Retries will get more attention than just one exception.